### ISSUE TITLE
Docker selenium grid not completing tests executions with latest versions

### ISSUE THREAD

**Issue by DevinPenny** [SeleniumHQ/docker-selenium#860]
I have a selenium grid server using docker containers for the hub/nodes, when I execute tests they appear to time out on the server. When I look at the hub page I can see that nodes are listed as in use, that is what is confusing to me.

The tests work as expected when I run them locally, as well as if I setup a local grid server using docker containers with the same latest versions. When I look at the logs from the grid server, it looks as if they just stop responding. I suspect there is a networking issue between the hub and nodes, but I am trying to determine the best way to troubleshoot my issue.

Log info: (you see it just stops at the last line, no more logs occur and the test times out)

chrome_3   | 13:13:23.381 INFO [SelfRegisteringRemote.registerToHub] - Registering the node to the hub: http://hub:4444/grid/register
chrome_3   | 13:13:23.411 INFO [SelfRegisteringRemote.registerToHub] - The node is registered to the hub and ready to use
chrome_3   | 13:15:36.690 INFO [ActiveSessionFactory.apply] - Capabilities are: {
chrome_3   |   "acceptSslCerts": true,
chrome_3   |   "browserName": "chrome",
chrome_3   |   "chromeOptions": {
chrome_3   |     "args": [
chrome_3   |       "-window-size=1366,768"
chrome_3   |     ]
chrome_3   |   },
chrome_3   |   "cleanSession": true,
chrome_3   |   "goog:chromeOptions": {
chrome_3   |   },
chrome_3   |   "javascriptEnabled": true,
chrome_3   |   "takesScreenshot": true
chrome_3   | }
chrome_3   | 13:15:36.691 INFO [ActiveSessionFactory.lambda$apply$11] - Matched factory org.openqa.selenium.grid.session.remote.ServicedSession$Factory (provider: org.openqa.selenium.chrome.ChromeDriverService)
chrome_3   | Starting ChromeDriver 2.45.615279 (12b89733300bd268cff3b78fc76cb8f3a7cc44e5) on port 21553
chrome_3   | Only local connections are allowed.
chrome_3   | 13:15:39.184 INFO [ProtocolHandshake.createSession] - Detected dialect: OSS
chrome_3   | 13:15:39.571 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session 678588466128ef6207f1caff6dfdec6c (org.openqa.selenium.chrome.ChromeDriverService)

(this is where the logs end)

Docker compose file:

version: '2.2'
services:
  hub:
    user: root
    image: "x.x.x.x/default-project/selenium/hub"
    ports:
      - "4444:4444"
    environment:
    - GRID_MAX_SESSION=48
    - TZ=America/New_York
  chrome:
    user: root
    cpu_count: 2
    image: "x.x.x.x/default-project/selenium/node-chrome"
    environment:
      - HUB_PORT_4444_TCP_ADDR=hub
      - HUB_PORT_4444_TCP_PORT=4444
      - NODE_MAX_INSTANCES=3
      - NODE_MAX_SESSION=3
      - TZ=America/New_York
    links:
      - "hub"
    expose:
      - "5555"
  firefox:
    user: root
    cpu_count: 2
    image: "x.x.x.x/default-project/selenium/node-firefox"
    environment:
      - HOME=/home/seluser
      - HUB_PORT_4444_TCP_ADDR=hub
      - HUB_PORT_4444_TCP_PORT=4444
      - NODE_MAX_INSTANCES=3
      - NODE_MAX_SESSION=3
      - TZ=America/New_York
    links:
      - "hub"
    expose:
      - "5555"

**Comment by diemol** [comment]
Hi @DevinPenny,

I am sorry, but there is no way to realise if something is going wrong after reading the issue. Could you please create a new issue and fill out the complete template? (Which requires all env information and a script to reproduce the issue).

**Comment by GeorgeMac** [comment]
P.s. I tried older versions of docker selenium `3.14.0-*` as well and got the same effect.
Which worries me even more 😂Sounds like something in common between my and @DevinPenny environment.

**Comment by bastien612** [comment]
I had the same issue when I deployed selenium/hub:3.141.59 and selenium/node-chrome-debug:3.141.59.
"driver.get(url)" was pending (actually I had a time out) when I run cucumber tests (written with the js webdriver) with "yarn test".
However it was completing when running a vanilla javascript selenium test with "node seleniumTest.js".

If someone could investigate, I can share a little project which should allow you to reproduce the issue.

Solution : I changed my base images version to **3.14.0-krypton** for selenium/hub and selenium/node-chrome-debug and it worked again.
I am not sure why...
